src/config/mod.rs

```diff
@@ -77,107 +77,39 @@ impl Config {
-            }
-            if let Some(val) = custom.readme_path {
-                self.readme_path = val;
-            }
-            if let Some(val) = custom.repository {
-                self.repository = Some(val);
-            }
-            if let Some(val) = custom.analytics {
-                self.analytics = Some(val);
-            }
-            if let Some(val) = custom.additional_pages {
-                self.additional_pages = Some(val);
-            }
-            if let Some(val) = custom.social {
-                self.social = Some(val);
-            }
-            if let Some(artifacts) = custom.artifacts {
-                // This value gets merged in a more fine-grain matter
-                // to allow earlier layers to set some values
-                if let Some(val) = artifacts.cargo_dist {
-                    self.artifacts.cargo_dist = Some(val);
-                }
-                if let Some(val) = artifacts.package_managers {
-                    self.artifacts.package_managers = Some(val);
-                }
-            }
-            if let Some(val) = custom.styles {
-                self.styles = val;
-            }
-            if let Some(val) = custom.version {
-                self.version = Some(val);
-            }
-            if let Some(val) = custom.logo {
-                self.logo = Some(val);
-            }
-            if let Some(val) = custom.favicon {
-                self.favicon = Some(val);
-            }
-            if let Some(val) = custom.path_prefix {
-                self.path_prefix = Some(val);
-            }
-            if let Some(val) = custom.changelog {
-                self.changelog = val;
-            }
-
-            match custom.mdbook {
-                Some(BoolOr::Val(val)) => {
-                    self.mdbook = Some(val);
-                }
-                Some(BoolOr::Bool(false)) => {
-                    // Disable mdbook support
-                    self.mdbook = None;
-                }
-                None | Some(BoolOr::Bool(true)) => {
-                    // Do nothing, use the previous value
-                    //
-                    // (Arguably "true" should mean something like Some(default)
-                    // but that's already the default and we don't want to clobber
-                    // other layers if they have an opinion.)
-                }
-            }
+            self.description.apply_val(custom.description);
+            self.dist_dir.apply_val(custom.dist_dir);
+            self.static_dir.apply_val(custom.static_dir);
+            self.homepage.apply_opt(custom.homepage);
+            self.name.apply_val(custom.name);
+            self.readme_path.apply_val(custom.readme_path);
+            self.repository.apply_opt(custom.repository);
+            self.analytics.apply_layer(custom.analytics);
+            // FIXME: should this get merged with e.g. `extend?`
+            self.additional_pages.apply_opt(custom.additional_pages);
```

Comment:
where is the other place that additional pages could be specified?

Reply:
ditto